core/trino-main/src/main/java/io/trino/execution/SqlStage.java

```diff
@@ -334,6 +335,19 @@ public String toString()
         return stateMachine.toString();
     }
 
+    public String getDebugInfo()
```

Comment:
toString already exists, but maybe we can just extend it to include more info?

Reply:
I am a bit concerned that original toString() (which is very brief) may already be used in some contexts where extended one does not make much sense. 
Let me try to verify that.

Reply:
Looks like this is fine. I reused `toString()`. Not I made it `synchronized` (getDebugInfo() should have been too)